### PR TITLE
Add content on parallelism, channels, concurrency

### PR DESCRIPTION
**Pull request by ChanderG**
Add 3 new chapters with various examples.

Am pretty new to Nim, and spent all day searching and compiling this set of pointers. All of the examples work for me - but would be good if someone experienced can confirm if all of this is good to go.

### CODE REVIEW

File: content/content/channels.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  echo "Received msg: " & msg
+
+commChan.open()
+spawn recvMsg()
+spawn sendMsg()
+sync()
+```
+
+Note that all messages are deep copied when sent.
+
```
Comment (flaviut) [inline content/content/channels.md:61]: It'd make sense to have line 79 here.

File: content/content/concurrency.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+title: Concurrency
+---
+
+# Concurrency
+
+Concurrency is provided in Nim, using `async`/`await` syntax. This feature is powered by the `asyncdispatch` [module](https://nim-lang.org/docs/asyncdispatch.html).
```
Comment (flaviut) [inline content/content/concurrency.md:7]:
```suggestion
Concurrency is provided in Nim, using `async`/`await` syntax. This feature is powered by [the `asyncdispatch` module](https://nim-lang.org/docs/asyncdispatch.html).
```

File: content/content/concurrency.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Refer to the [docs](https://nim-lang.org/docs/asyncdispatch.html) for more info. See also, [this blog post](https://peterme.net/asynchronous-programming-in-nim.html).
+
+## Higher Async modules
+
+The underlying `asyncdispatch` modules is used by several higher level modules:
+
+1. [asyncfile](https://nim-lang.org/docs/asyncfile.html) for async file operations.
+2. [asyncnet](https://nim-lang.org/docs/asyncnet.html) for async networking operations.
+3. [asynchttpserver](https://nim-lang.org/docs/asynchttpserver.html) for an async http server.
```
Comment (flaviut) [inline content/content/concurrency.md:57]:
```suggestion
1. [`asyncfile`](https://nim-lang.org/docs/asyncfile.html) for async file operations.
2. [`asyncnet`](https://nim-lang.org/docs/asyncnet.html) for async networking operations.
3. [`asynchttpserver`](https://nim-lang.org/docs/asynchttpserver.html) for an async http server.
```

File: content/content/concurrency.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+In the example above, we have used `waitFor` on 2 async functions, so the execution blocks until both functions are run to completion.
+
+An alternative option is:
+``` nimrod
+runForever()
+```
+which blocks, waiting indefinitely for all asynchronous functions.
+
+Refer to the [docs](https://nim-lang.org/docs/asyncdispatch.html) for more info. See also, [this blog post](https://peterme.net/asynchronous-programming-in-nim.html).
```
Comment (flaviut) [inline content/content/concurrency.md:49]:
```suggestion
You may also find [Peter's article on async programming](https://peterme.net/asynchronous-programming-in-nim.html) helpful.
```

**Review by flaviut — APPROVED** [review]
Appreciate all your hard work here! It's really been a long time since I've worked with Nim, but things do look reasonable!

My comments below are mostly editorial and stylistic.

You'll want to add these pages [to the TOC](https://raw.githubusercontent.com/flaviut/nim-by-example/master/content/content/toc.md) as well.

File: content/content/parallelism.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+1. The experimental `parallel` statement.
+2. Return data from `spawn` calls.
+
+Details for both of these are documented [here](https://nim-lang.org/docs/manual_experimental.html#parallel-amp-spawn).
+
+Apart from these options, Nim also has:
+1. OpenMP support using the `||` operator. See the example [here](https://rosettacode.org/wiki/Concurrent_computing#Nim).
+2. [Coroutines](https://nim-lang.org/docs/coro.html).
+
+However, both of these are not ready yet for general usuage.
```
[the remark this answers is not in the document]
Comment (ChanderG) [reply content/content/parallelism.md:60]: I couldn't find any documentation on the OpenMP support and the `||` operator, except for that example on Rosetta code. Still, running that example led to Segfaults.

As for coroutines, the doc mentions is to be an Unstable API and I think I saw a comment somewhere about it not working on Mac?, not sure.

To be safe, we can drop this entire para.

File: content/content/channels.md (hunk 1 above)
Comment (ChanderG) [reply content/content/channels.md:61]: 79 depends on 70-77. If you mean moving 70-79 to before 62, I can do that.

Comment (flaviut) [reply content/content/channels.md:61]: Oh, I see. Looks fine the way it is!

**Comment by flaviut** [comment]
Looks great! Feel free to hit the big green button whenever you're ready.

**Pull request merged**